packages/account/src/BiconomySmartAccountV2.ts

```diff
@@ -141,18 +154,47 @@ export class BiconomySmartAccountV2 extends BaseSmartContractAccount {
    * @throws An error if something is wrong with the smart account instance creation.
    */
   public static async create(biconomySmartAccountConfig: BiconomySmartAccountV2Config): Promise<BiconomySmartAccountV2> {
-    const config = {
-      ...biconomySmartAccountConfig,
-    };
+    let chainId = biconomySmartAccountConfig.chainId;
+
+    // Signer needs to be initialised here before defaultValidationModule is set
+    if (biconomySmartAccountConfig.signer) {
+      const signer = biconomySmartAccountConfig.signer;
+      const isViemWalletClient = !(signer instanceof WalletClientSigner);
+      if (isViemWalletClient) {
```

Comment:
What happens if it's not ViemWalletClient?

Reply:
The regular flow, it passes through unchanged. Currently the default expectation is that the signer is AA's WalletClientSigner [(see here)](https://github.com/bcnmy/biconomy-client-sdk/blob/0f0a1b6b4b5d17b1d5bba13d285607881fcb8aed/packages/account/tests/account.spec.ts#L29)